module/VuFind/src/VuFind/Controller/AuthorityController.php

```diff
@@ -58,31 +58,20 @@ public function __construct(ServiceLocatorInterface $sm)
      */
     public function homeAction()
     {
-        // If we came in with a record ID, forward to the record action:
+        // If we came in with a record ID, forward to the record action; this
+        // provides backward compatibility with multiple legacy routes.
         if ($id = $this->params()->fromRoute('id', false)) {
-            $this->getRequest()->getQuery()->set('id', $id);
-            return $this->forwardTo('Authority', 'Record');
+            if ($id === 'Record') {
+                $queryId = $this->params()->fromQuery('id');
+                $id = $queryId === null ? $id : $queryId;
```

Comment:
These two lines can be simplified as `$id = $this->params()->fromQuery('id', $id);`, right?

Reply:
Sorry, I meant replacing just lines 65 and 66 as:

```
            if ($id === 'Record') {
                $id = $this->params()->fromQuery('id', $id);
            }

```

Reply:
Ahh, sorry for misunderstanding; I believe you are correct and will fix it now...